app/components/App.js

```diff
@@ -1,329 +1,307 @@
     });
-};
+  }
+
 
-    
-// some room for consolidation of functions wiht check in the title. checkListSame is the most general
-checkDuplicate(details, classType) {
+  // some room for consolidation of functions wiht check in the title. checkListSame is the most general
+  checkDuplicate(details, classType) {
     //do stuff
-    var duplicateStyle='notDuplicate'
-    let ind=[]
-    let isInList
-    
-    isInList = this.checkListSame(details,3)
-    if (isInList[0]) {duplicateStyle='duplicateCSS'}
-
-    
-    if (classType =='group') {
-      let ind
-      let bad = []
-      let affectedOrders=[]
-        details.value.map((orderable,index) => {
-            ind = this.state.duplicateOrders.findIndex((x => x.id == orderable.id))
-            if (ind !=-1) {bad[index] = orderable}
-            return (bad)
-            }) 
-                  
-         //all the items in a group are affected
-        if(bad.length == details.value.length) {
-                    duplicateStyle='duplicateCSS'
+    let duplicateStyle = 'notDuplicate';
+    const ind = [];
+    let isInList;
+
+    isInList = this.checkListSame(details, 3);
+    if (isInList[0]) {duplicateStyle = 'duplicateCSS';}
+
+
+    if (classType === 'group') {
+      let ind;
+      const bad = [];
+      const affectedOrders = [];
+      details.value.map((orderable, index) => {
+        ind = this.state.duplicateOrders.findIndex((x => x.id === orderable.id));
+        if (ind !== -1) {bad[index] = orderable;}
+        return (bad);
+      });
+
+      //all the items in a group are affected
+      if (bad.length === details.value.length) {
+        duplicateStyle = 'duplicateCSS';
         // some of the items in a group are affected
-        } else if(bad.length != 0) {
-            duplicateStyle='notDuplicate'
+      } else if (bad.length !== 0) {
+        duplicateStyle = 'notDuplicate';
         // the group is not affected
-        } else {duplicateStyle='notDuplicate'}
+      } else {duplicateStyle = 'notDuplicate';}
     }
-    return(duplicateStyle)
-}
+    return (duplicateStyle);
+  }
 
-// some room for consolidation of functions wiht check in the title. checkListSame is the most general
-checkWarning(details, classType) {
+  // some room for consolidation of functions wiht check in the title. checkListSame is the most general
+  checkWarning(details, classType) {
     //currently this function only exists so that you know which list to check
-    var index=[]
-    var isInList
-    isInList = this.checkListSame(details,2)
-        
-    return(isInList[0])
-}
+    const index = [];
+    let isInList;
+    isInList = this.checkListSame(details, 2);
+
+    return (isInList[0]);
+  }
+
+  // looks to see if an object is contained in a list of the same class of objects
+  checkListSame(object, listType, errorlocation) {
+    let index = -1;
+    let isInList = false;
+    let len = 0;
+    let list;
+    let obj = {};
+
+
+    // there is probably a better way to do this
```

Comment:
There is, but what I'd focus more on is why you're doing this. The comment says "looks to see if an object is contained in a list of the same class of objects", but why is it a method in this component? I suspect there's a better place for this, but more broadly... 

Part of what is throwing me off here is the use of listType being a number to represent a listType. I'd recommend a string, symbol, variable, any number of other possibilities.